### PR TITLE
Merge CoreCLR-based official builds legs into one leg per platform

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Today, we split our official builds for CoreCLR-based artifacts into multiple separate jobs and we flow artifacts between the jobs. This adds significant complexity to our build and takes longer due to waiting for machine provisioning. Additionally, for the VMR, we need to build each platform in a single job as we need to be able to build all of the constituent repos in one job.

This PR merges each CoreCLR platform to build in a single job. It continues to keep one job for the CrossDac packages, as those need to be built cross-machine. Additionally, it preserves the Libraries AllConfigurations leg as a separate job as that is platform-agnostic.

Various platform legs for CoreCLR have additional pre and post build steps to account for signing (DAC/DBI and macOS tools), entitlements (macOS), uploading artifacts for the CrossDac job (Linux), and building installers in specific containers (Linux).

> [!NOTE]
> Official build link: https://dev.azure.com/dnceng/internal/_build/results?buildId=2280331&view=results

> [!IMPORTANT]
> Due to branch restrictions on the service connection for the DAC/DBI signing, I was unable to validate it end-to-end.

### CODE REVIEW

File: eng/pipelines/runtime-official.yml [hunk 1]
```diff
@@ -59,46 +59,236 @@ extends:
...
+                parameters:
+                  buildArgs: -s clr.corelib+clr.nativecorelib+clr.nativeaotlibs+clr.tools+clr.packages+libs+host+packs -c $(_BuildConfig)
+
+              # Upload the results.
+              - template: /eng/pipelines/common/upload-intermediate-artifacts-step.yml
+                parameters:
+                  name: $(osGroup)$(osSubgroup)_$(archType)
+
+      #
+      # Build CoreCLR
```
Comment (ViktorHofer) [inline eng/pipelines/runtime-official.yml:174]: This is a bit misleading, as this step builds not just CoreCLR but also libs, host and packages. Might be good to refine that comment.

**Comment by jkoritzinsky** [comment]
DAC/DBI signing has been validated.

File: eng/pipelines/runtime-official.yml [hunk 2]
```diff
@@ -58,47 +58,243 @@ extends:
...
+            postBuildSteps:
+            - template: /eng/pipelines/coreclr/templates/sign-diagnostic-files.yml
+              parameters:
+                basePath: $(Build.SourcesDirectory)/artifacts/bin/coreclr
+                isOfficialBuild: ${{ variables.isOfficialBuild }}
+                timeoutInMinutes: 30
+            # Now that we've signed the diagnostic files, do the rest of the build.
+            - template: /eng/pipelines/common/templates/global-build-step.yml
+              parameters:
+                buildArgs: -s clr.corelib+clr.nativecorelib+clr.nativeaotlibs+clr.tools+clr.packages+libs+host+packs -c $(_BuildConfig)
```
Comment (hoyosjs) [inline eng/pipelines/runtime-official.yml:86]: Global-build-step should probably get a displayName parameter so it's easier to understand the global build post steps.

File: eng/pipelines/runtime-official.yml [hunk 3]
```diff
@@ -485,9 +633,9 @@ extends:
             - Build_tvossimulator_arm64_release_AllSubsets_Mono
             - Build_tvossimulator_x64_release_AllSubsets_Mono
             - Build_windows_x64_release_CrossAOT_Mono
-            - installer_coreclr__windows_x64_Release_
-            - installer_coreclr__windows_x86_Release_
-            - installer_coreclr__windows_arm64_Release_
+            - Build_windows_x64_release_AllSubsets_CoreCLR
```
Comment (hoyosjs) [inline eng/pipelines/runtime-official.yml:636]: Now that we build like this, I don't think All subsets in the name really helps, it just lengthens the name and AzDO has a pretty narrow column for it. Can we just drop it?

Comment (ViktorHofer) [reply eng/pipelines/runtime-official.yml:636]: Agreed - it should be the default and legs that don't follow this pattern should have a display name describing their scenario.

File: src/coreclr/build-runtime.cmd [hunk 4]
```diff
@@ -211,8 +211,13 @@ if NOT "%__BuildType%"=="Release" (
...
 
-set "__BinDir=%__RootBinDir%\bin\coreclr\%__TargetOS%.%__TargetArch%.%__BuildType%"
-set "__IntermediatesDir=%__RootBinDir%\obj\coreclr\%__TargetOS%.%__TargetArch%.%__BuildType%"
+set __TargetOSDirName=%__TargetOS%
+if "%__TargetOS%"=="alpine" (
+    set __TargetOSDirName=linux_musl
+)
+
+set "__BinDir=%__RootBinDir%\bin\coreclr\%__TargetOSDirName%.%__TargetArch%.%__BuildType%"
+set "__IntermediatesDir=%__RootBinDir%\obj\coreclr\%__TargetOSDirName%.%__TargetArch%.%__BuildType%"
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply src/coreclr/build-runtime.cmd:220]: This is purely for the alpine DAC. The Alpine Cross-DAC is the only place where we need to differentiate between linux and linux_musl in our output directories.

**Comment by jkoritzinsky** [comment]
Official build link for [`9cebeb1` (#92901)](https://github.com/dotnet/runtime/pull/92901/commits/9cebeb18d1b463ddc8468fde36b7b41e68a20a3e): https://dev.azure.com/dnceng/internal/_build/results?buildId=2283609&view=results

**Comment by jkoritzinsky** [comment]
Official build link for [`9e64389` (#92901)](https://github.com/dotnet/runtime/pull/92901/commits/9e64389f227e379b80951c196bcd5a784e26a826): https://dev.azure.com/dnceng/internal/_build/results?buildId=2283677&view=results

**Review by agocke — APPROVED** [review]
LGTM, although I'm not sure I completely understand the crossdac interaction

**Comment by jkoritzinsky** [comment]
@hoyosjs can you take another look at the CrossDac work and approve/request changes? I think I've got it all right, but I really want to make sure I don't break that space.

**Review by hoyosjs — APPROVED** [review]

**Comment by lewing** [comment]
Looks like this broke the official build by producing packages with -ci version

**Pull request merged**